src/redux/reducers/userReducer.js

```diff
@@ -0,0 +1,18 @@
+import createReducer from './createReducer';
+import { signInUserSuccess, signInUserError } from '../actions/userActions';
+
+const intialState = {
+  value: undefined,
+  auth: undefined,
+  error: undefined
+};
+
+export default createReducer(intialState, {
+  [signInUserSuccess]: (state, { payload }) => ({
+    ...state,
+    value: payload.data,
+    auth: payload.headers,
+    error: undefined
+  }),
```

Comment:
Since you are not using any other fields you can just return a new state without even checking the previous one.
```suggestion
  [signInUserSuccess]: (state, { payload: { data, headers } }) => ({
    value: data,
    auth: headers
  }),
```